packages/near-fast-auth-signer/src/components/SignMultichain/SignMultichain.tsx

```diff
@@ -138,6 +181,9 @@ function SignMultichain() {
           setValid(true);
           setIsDomainKey(transaction?.domain === event?.origin);
           setMessage(transaction);
+          if (transaction?.domain === event?.origin && !isSafariBrowser) {
```

Comment:
Duplicated `if (transaction?.domain === event?.origin && !isSafariBrowser) {`

I think one approach we can take it's hide the modal until we need to display it. 

So, instead of notify when we need to hide the modal, we notify the caller when we need to display it. 

What do you think? So we can do all the operations inside a single `if`

Reply:
I tried and I ended up duplicating the if statement because if I use it under single if statement, it does show the modal for some period of time (fairly broken) until the rest begins. 

To give you a bit more context, there are two logic to consider:
1. UI hide logic
2. Executing `signMultichainTransaction` 

The reason why I didn't put them in single if statement is because there is async operation of fetching prices before calling `signMultichainTransaction`. So if I wrap them both under one if statement and if it is located after price fetch request, there is a broken UI happens. I can't put if statement with both logic before price fetch because it won't work at all. 

So it may look like if statement is duplicated, but the way I see it is that it is optimised for UX.